src/message.js

```diff
@@ -543,20 +549,28 @@ export class Message {
       const signingKeyID = signingKeyIDs[signingKeys.length - 1 - i];
       const signingKey = await primaryKey.getSigningKey(signingKeyID, date, userIDs, config);
       const onePassSig = new OnePassSignaturePacket();
+      onePassSig.version = signingKey.keyPacket.version === 6 ? 6 : 3;
       onePassSig.signatureType = signatureType;
       onePassSig.hashAlgorithm = await getPreferredHashAlgo(primaryKey, signingKey.keyPacket, date, userIDs, config);
       onePassSig.publicKeyAlgorithm = signingKey.keyPacket.algorithm;
       onePassSig.issuerKeyID = signingKey.getKeyID();
+      if (onePassSig.version === 6) {
+        onePassSig.issuerFingerprint = signingKey.keyPacket.fingerprint;
+        onePassSig.generateSalt();
+      }
       if (i === signingKeys.length - 1) {
         onePassSig.flags = 1;
       }
       return onePassSig;
     })).then(onePassSignatureList => {
-      onePassSignatureList.forEach(onePassSig => packetlist.push(onePassSig));
+      onePassSignatureList.forEach(onePassSig => {
+        packetlist.push(onePassSig);
+        signatureSalts.push(onePassSig.salt);
+      });
     });
 
     packetlist.push(literalDataPacket);
-    packetlist.push(...(await createSignaturePackets(literalDataPacket, signingKeys, signature, signingKeyIDs, date, userIDs, notations, false, config)));
+    packetlist.push(...(await createSignaturePackets(literalDataPacket, signingKeys, signature, signingKeyIDs, date, userIDs, notations, signatureSalts, false, config)));
```

Comment:
We could generate the salt when generating the signature packet instead, and then create the one-pass signature packet from the signature packet (though we could also do that as a separate change)